dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -183,19 +183,36 @@ if (isMaster) {
                 shell.openExternal(url);
             });
 
-            const saveState = () => {
-                const bounds = newWindow.getBounds();
-                electronStore.set(windowStateName, {
-                    isMaximized: newWindow.isMaximized(),
-                    width: bounds.width,
-                    height: bounds.height,
-                    x: bounds.x,
-                    y: bounds.y
-                })
-            }
-            newWindow.on('close', saveState);
-            newWindow.on('resize', saveState);
-            newWindow.on('move', saveState);
+            // Save the window geometry state on every change
+            const saveWindowState = () => {
+                try {
+                    let bounds;
+                    if (newWindow.isMaximized()) {
+                        bounds = electronStore.get(WINDOW_STATE, {});
+                    } else {
+                        bounds = newWindow.getBounds();
+                    }
+                    electronStore.set(WINDOW_STATE, {
+                        isMaximized: newWindow.isMaximized(),
+                        width: bounds.width,
+                        height: bounds.height,
+                        x: bounds.x,
+                        y: bounds.y
+                    });
+                } catch (e) {
+                    console.error("Error while saving window state.", e);
+                }
+            };
+            let delayedSaveTimeout;
+            const saveWindowStateDelayed = () => {
+                if (delayedSaveTimeout) {
+                    clearTimeout(delayedSaveTimeout);
+                }
+                delayedSaveTimeout = setTimeout(saveWindowState, 1000);
```

Comment:
Why did you choose 1000ms for debouncing? As a user, I find such a delay extremely long (and would prefer something faster like 200ms).

Reply:
Yes, it happens in the background. The delay should be long enough to avoid file system problems on Windows.